18F/identity-idp
# LG-11588: Don't tell users to get another letter when they can't

## 🎫 Ticket

Link to the relevant ticket:
[LG-11588](https://cm-jira.usa.gov/browse/LG-11588)

## 🛠 Summary of changes

When users enter an incorrect verification code in the verify by mail code, the error message suggests that they "Please request another letter for a new code."

This PR updates that message to remove that suggestion **if** the user cannot currently request another GPO letter (either because their most recent request was too recent or they have requested the maximum number of letters). The message is unchanged for users who are still able to request letters.

This change involved adding new translation keys, using adapted versions of the text we already had:

| Language | Text |
| -- | -- |
| English | Your verification code has expired. <del>Please request another letter for a new code.</del> |
| Spanish | Tu código de verificación ha caducado. <del>Vuelve a solicitar otra carta para recibir un nuevo código. </del> |
| French | Votre code de vérification a expiré. <del>Veuillez solliciter une autre lettre afin d’obtenir un nouveau code.</del> |
| Chinese | 你的验证码已过期。<del>请要求另外一封信来得到新代码。</del> |

## 📜 Testing Plan

First, configure your application.yml to test the "must wait 1 day" condition:

```yaml
usps_confirmation_max_days: -1                             # Expire GPO codes by default
max_mail_events: 4                                         # Max total of 4 letter requests
minimum_wait_before_another_usps_letter_in_hours: 24       # Wait 1 day before you can request another letter
```

1. Go through identity verification, requesting a new GPO code
1. Enter the code, and verify you receive the error message "Your verification code has expired."

Now, configure your application.yml to test the "requested too many letters" condition:

```yaml
usps_confirmation_max_days: -1                             # Expire GPO codes by default
max_mail_events: 1                                         # Max total of 1 letter requests
minimum_wait_before_another_usps_letter_in_hours: 0        # No wait before requesting another
```

1. Go through identity verification, requesting a new GPO code
1. Enter the code, and verify you receive the error message "Your verification code has expired."

Now configure your application.yml such that the user will not be prevented from requesting a new letter:

```yaml
usps_confirmation_max_days: -1                             # Expire GPO codes by default
max_mail_events: 4                                         # Max total of 4 letter requests
minimum_wait_before_another_usps_letter_in_hours: 0        # No wait before requesting another
```

1. Go through identity verification, requesting a new GPO code
1. Enter the code, and verify you receive the error message "Your verification code has expired. Please request another letter for a new code."

## 👀 Screenshots

If relevant, include a screenshot or screen capture of the changes.

<details>
<summary>Before:</summary>

![verify-by_mail-enter_code-en-BEFORE](https://github.com/18F/identity-idp/assets/364697/cbb2e315-d7eb-4ca3-aaf7-284446f90cca)


</details>

<details>
<summary>After:</summary>

![verify-by_mail-enter_code-en-AFTER](https://github.com/18F/identity-idp/assets/364697/0c4664f9-f25f-42f8-b34d-371f8da63313)


</details>
